### DISCUSSION TITLE
Connection sensor with mqtt server

### DISCUSSION THREAD

**Discussion by MilorES** [firstof9/openevse#223] [Ideas]
It would be possible to put it like this if the car is not charging restart the openevse, I understand that there is a service or you can press the button that is there, right?

**Comment by firstof9** [comment]
Yes there are button entities you can automate on to restart the EVSE and/or the wifi module.

**Comment by MilorES** [comment]
Could an identity be set to detect if the MQTT is connected or not?

This is what it looks like on the web:
![imagen](https://github.com/firstof9/openevse/assets/7502697/630b742d-0171-4cd5-8159-be766436a23b)

This would be cool if it was in HA:
![imagen](https://github.com/firstof9/openevse/assets/7502697/1fa42afe-507b-4cea-b430-896dd0feb514)
*In this case the states do not match.

If I had that identity, I could restart the WIFI module.

Note: I never tried to "press" a button from the automations... only calls to services.

**Reply by firstof9** [reply]
Should be able to do that.

**Comment by MilorES** [comment]
That sensor does not appear in the integration... only on the OpenEVSE web page. Will you put it?

Or do you mean that I can somehow create it myself?
If so, I would not know where to start, any recommendations?

**Reply by firstof9** [reply]
I'll see about adding it to the next release.
Thanks!

**Reply by firstof9** [reply]
If you'd like to give it a test you can grab the files from the mqtt branch and give it a whirl.

https://github.com/firstof9/openevse/tree/mqtt-sensor

**Reply by MilorES** [reply]
Perfect!
![imagen](https://github.com/firstof9/openevse/assets/7502697/60f03982-0bed-4bd9-b5ea-0c4add699b5c)
If you go through Figueres (Spain) you are invited to coffee, beer or drink!

Thank you so much

Nota: I'm a little late! Because it was already in the MAIN
